oauthproxy.go

```diff
@@ -57,17 +56,8 @@ type allowedRoute struct {
 
 // OAuthProxy is the main authentication proxy
 type OAuthProxy struct {
-	CookieSeed     string
-	CookieName     string
-	CSRFCookieName string
-	CookieDomains  []string
-	CookiePath     string
-	CookieSecure   bool
-	CookieHTTPOnly bool
-	CookieExpire   time.Duration
-	CookieRefresh  time.Duration
-	CookieSameSite string
-	Validator      func(string) bool
+	CookieOptions *options.Cookie
```

Comment:
Oooh! I like this :D

Reply:
The first step in a long journey of figuring out what the heck is going on in the `OAuthProxy` struct -- and why some fields are public and others private 😅 

And why other fields even exist at all.